src/utils.rs

```diff
@@ -1,21 +1,92 @@
-use multiaddr::Multiaddr;
+use multiaddr::{Multiaddr, Protocol};
+use secio::PeerId;
 use std::net::SocketAddr;
 
 /// Change multiaddr to socketaddr
 pub fn multiaddr_to_socketaddr(addr: &Multiaddr) -> Result<SocketAddr, ()> {
-    use multiaddr::Protocol;
+    let mut iter = addr.iter().peekable();
 
+    while iter.peek().is_some() {
+        match iter.peek() {
```

Comment:
Use a more robust way to extract ip and addr
``` rust
let mut ip = None;
let mut port = None;
for item in iter {
    match item {
        Protocol::Ip4(ip4) => ip = Some(IpAddr::V4(ip4)),
        Protocol::Ip6(ip6) => ip = Some(IpAddr::V6(ip6)),
        Protocol::Tcp(tcp_port) => port = Some(tcp_port)
    }
    if ip.is_some() && port.is_some() {
        break;
    }
}
```

Reply:
I think the method of the iterator I used is not a problem, you can look at the test.

I don't want to have something like this:
```
/ip4/127.0.0.1/p2p/fdafdafdsafdsfds/tcp/1337
```